### DISCUSSION TITLE
Welcome to obs_weather_time_overlay Discussions!

### DISCUSSION THREAD

**Discussion by ngholson** [ngholson/obs_weather_time_overlay#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

default post...

**Comment by rickscully** [comment]
I would like to be able to display both imperial and metric. I could easily just duplicate the script I assume, but I am unsure how to display an F or a C after the º. 

I also would like to put a | between the temp and the time when using simple. Is that possible?

Any advice appreciated. Thanks!

**Comment by ngholson** [comment]
Yes it is. I will see if I can help you when I get home, about an hour or
so.

**Comment by ngholson** [comment]
the lines of code you would want to edit is below, they are lines 112-115
in the html file

                            if (weatherDisplay.toLowerCase() == "simple") {
                                if (weatherIcons == 1) { document.
getElementById("id-weather").innerHTML = `${curtemp}&deg; <img
style='height: ${iconHeight};' src=${wimg}> ${time}`; return; }
                                document.getElementById("id-weather").
innerHTML = `${curtemp}&deg;  ${time}`; return;
                            }

**Reply by rickscully** [reply]
Thank you very much.

**Comment by ngholson** [comment]
this may solve your issue. just copy and paste all of this to replace the
everything in between the <body></body> tags

<body>
    <script>
    var wVer = 101;
    if (weatherVersion > wVer && localStorage.getItem(wVer+"."+
weatherVersion+"update") != 1 && autoCheckUpdates == true) {
        if (confirm("g4Weather v"+weatherFriendly+" Update Available.
Please visit:\nhttps://github.com/ngholson/obs_weather_time_overlay\nClick
'CANCEL' to hide future notifications about this update") != true){
        localStorage.setItem(wVer+"."+weatherVersion+"update",1);
        }
    }
    </script>
    <center>
        <span class="weather-box" id="id-weather"></span><br>
    </center>
    <script>
        var language = window.navigator.userLanguage || window.navigator.
language;
        const regionTranslate = new Intl.DisplayNames([language], { type:
'region' });
        document.querySelector(':root').style.setProperty('--txtSize',
textSize); document.querySelector(':root').style.setProperty('--txtColor',
textColor);
        document.querySelector(':root').style.setProperty('--wBack',
weatherBackground);
        document.querySelector(':root').style.setProperty('--wBorder',
weatherBorder+"px");
        refreshData();
        function weatherCheck() {
            var time = time24hour != true ? currentTime(new Date()) : time24
(new Date());
            $.ajax({
                url: 'https://api.openweathermap.org/data/2.5/weather',
                data: { units: `${yourUnits}`, q: `${yourCity}`, appid: `${
yourApiKey}`, lang: `${language}` },
                dataType: 'json',
                success: function (apiResponse) {
                    if (dynamicBackground == 1) {
                        if (apiResponse.weather[0].icon.slice(2) == "n") {
document.querySelector(':root').style.setProperty('--wBack',
"linear-gradient(to
bottom, black , dimgrey"); document.querySelector(':root').style.setProperty
('--txtColor', "white"); }
                        else { document.querySelector(':root').style.
setProperty('--wBack', "linear-gradient(to bottom, skyblue ," +
weatherBackground); document.querySelector(':root').style.setProperty(
'--txtColor', "black"); }
                    }

                    var curtemp = Math.round(apiResponse.main.temp);
                    let secondTemp = Math.round(((apiResponse.main.temp)-32
)*5)/9;
                    var wimg = "./images/pack" + `${iconPack}` + "/" +
apiResponse.weather[0].icon + ".png";
                    $.ajax({
                        url: 'http://api.openweathermap.org/geo/1.0/direct',
                        data: { q: `${yourCity}`, appid: `${yourApiKey}` },
                        dataType: 'json',
                        success: function (geoResponse) {
                            if (geoResponse[0].country.toLowerCase() != "us")
{
                                if (geoResponse[0].state == apiResponse.name
|| geoResponse[0].state == null) {
                                    geoResponse[0].state = regionTranslate.
of(geoResponse[0].country);
                                }
                            }
                            if (weatherDisplay.toLowerCase() == "temp") {
                                if (weatherIcons == 1) { document.
getElementById("id-weather").innerHTML = `<img style='height: ${iconHeight};'
src=${wimg}> ${curtemp}&deg;`; return; }
                                document.getElementById("id-weather").
innerHTML = `${curtemp}&deg;`; return;
                            }
                            if (weatherDisplay.toLowerCase() == "time") {
document.getElementById("id-weather").innerHTML = `${time}`; return; }
                            if (weatherDisplay.toLowerCase() == "simple") {
                                if (weatherIcons == 1) { document.
getElementById("id-weather").innerHTML = `${curtemp}&deg; <img
style='height: ${iconHeight};' src=${wimg}> ${time}`; return; }
                                document.getElementById("id-weather").
innerHTML = `${curtemp}&deg;F / ${secondTemp}&deg;C | ${time}`; return;
                            }
                            if (weatherDisplay.toLowerCase() == "weather") {
                                if (weatherIcons == 1) {
                                    document.getElementById("id-weather").
innerHTML = `${geoResponse[0].name}, ${geoResponse[0].state}: ${curtemp}&deg;
<img style='height: ${iconHeight};' src=${wimg}> ${apiResponse.weather[0].
description}`;
                                } else { document.getElementById(
"id-weather").innerHTML = `${geoResponse[0].name}, ${geoResponse[0].state}:
${curtemp}&deg; \/ ${apiResponse.weather[0].description}`; }
                                return;
                            }
                            //start of full
                            if (weatherIcons == 1) {
                                document.getElementById("id-weather").
innerHTML = `${geoResponse[0].name}, ${geoResponse[0].state}: ${curtemp}&deg;
<img style='height: ${iconHeight};' src=${wimg}> ${apiResponse.weather[0].
description} ${clockSeperator} ${time}`;
                            } else { document.getElementById("id-weather").
innerHTML = `${geoResponse[0].name}, ${geoResponse[0].state}: ${curtemp}&deg;
\/ ${apiResponse.weather[0].description} ${clockSeperator} ${time}`; }
                        }
                    });
                }
            });
        }
        function currentTime(date) {
            var hours = date.getHours();
            var minutes = date.getMinutes();
            var isAmPm = hours >= 12 ? 'PM' : 'AM';
            hours = hours % 12;
            hours = hours ? hours : 12;
            minutes = minutes < 10 ? '0' + minutes : minutes;
            var curTime = hours + ':' + minutes + ' ' + isAmPm;
            return curTime;
        }
        function time24(date) {
            var hours = date.getHours();
            var minutes = date.getMinutes();
            hours = hours < 10 ? '0' + hours : hours;
            minutes = minutes < 10 ? '0' + minutes : minutes;
            var curTime24 = hours + ':' + minutes;
            return curTime24;
        }
        function refreshData() {
            weatherCheck();
            setTimeout(refreshData, 60000);
        }
    </script>
</body>

**Comment by ngholson** [comment]
of course i am assuming that you are not using the weather icons.

**Comment by ngholson** [comment]
output should look similar to this
[image: image.png]

**Comment by ngholson** [comment]
actually add this code
secondTemp= Math.round(secondTemp);

below this line
let secondTemp = Math.round(((apiResponse.main.temp)-32)*5)/9;
